{{cookiecutter.repo_name}}/templates/cpp/{{cookiecutter.repo_name}}.cpp

```diff
@@ -2,18 +2,44 @@
 #include <mpi.h>
 #include <stdexcept>
 #include <string.h>
+#include <map>
+#include <memory>
 #include "mdi.h"
+
+typedef std::map<std::string, MDI_Comm> EngineMap;
+
+EngineMap connect_to_engines(int nengines) {
+  
+  EngineMap engines;
+
+  for (int iengine=0; iengine < nengines; iengine++) {
+
+    MDI_Comm comm;
+    MDI_Accept_communicator(&comm);
  
-using namespace std;
+    // Determine the name of this engine
+    std::unique_ptr<char[]> engine_name(new char[MDI_NAME_LENGTH]);
+    MDI_Send_command("<NAME", comm);
+    MDI_Recv(engine_name.get(), MDI_NAME_LENGTH, MDI_CHAR, comm);
  
-int main(int argc, char **argv) {
- 
-  // Initialize the MPI environment
-  MPI_Comm world_comm;
-  MPI_Init(&argc, &argv);
+    std::cout << "Engine name: " << engine_name.get() << std::endl;
+
+    engines[engine_name.get()] = comm;
+
+  }
+
+  return engines;
+
+}
+
+void initialization(int *argc, char ***argv, MPI_Comm *mpi_world_comm) {
```

Comment:
Hi Jessica.  For clarity, I would suggest changing the names of these arguments to `argc_ptr`, `argv_ptr`, and `mpi_world_comm_ptr`, respectively.